### PR TITLE
Rename istio-ca service/image to citadel.

### PR DESCRIPTION
**Pull request by myidpt**

### CODE REVIEW

File: install/kubernetes/templates/istio-citadel-one-namespace.yaml.tmpl [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
-        istio: istio-ca
+        istio: citadel
       annotations:
         sidecar.istio.io/inject: "false"
     spec:
-      serviceAccountName: istio-ca-service-account
+      serviceAccountName: istio-citadel-service-account
       containers:
-      - name: istio-ca
-        image: {CA_HUB}/istio-ca:{CA_TAG}
```
Comment (incfly) [inline install/kubernetes/templates/istio-citadel-one-namespace.yaml.tmpl]: (Could be a TODO)

We may still have "CA" somewhere. Somewhere using "ca" makes sense like --self-signed-ca, some like here (CA_HUB) may should renamed to CITDADEL_HUB (similar to MIXER_HUB).

**Comment by myidpt** [comment]
@incfly For "citadel" and "istio-citadel", I'm following the same scheme used by Pilot and Mixer. Basically, the image name is "citadel", the service name is "istio-citadel", the label is "istio: citadel".

File: install/kubernetes/templates/istio-citadel-one-namespace.yaml.tmpl (hunk 1 above)
Comment (myidpt) [reply install/kubernetes/templates/istio-citadel-one-namespace.yaml.tmpl]: Yes, good catch. I'll modify them in another PR :)

**Comment by myidpt** [comment]
@incfly For the doc to clarify what "citadel" is, I will update istio.io pages :)

**Comment by kyessenov** [comment]
I think the current style is somewhat inconsistent. For example:
1. istio-telemetry service with label istio:mixer component
2. istio-policy service with label istio:mixer component
3. istio-pilot service with label istio:pilot

Maybe we should call this istio-ca service with istio: citadel component?

**Comment by myidpt** [comment]
@kyessenov We actually try to avoid "istio-ca" because the Istio CA component will not only be used as CA in the future, it can be used RA integrated with external CA, Istio identity provider, or internal JWT issuer. istio-ca will not be a precise name any more.

**Comment by myidpt** [comment]
@wattli Yes, PRs like this are not easy.

**Comment by kyessenov** [comment]
@myidpt yes, but you can create multiple services pointing to the same component like it is done with Mixer.

**Comment by myidpt** [comment]
@kyessenov I saw for istio: mixer label, there is an istio-mixer service, so we could do the same thing:
For istio: citadel label, there is an istio-citadel service, and maybe in the future, we will have istio-jwt-issuer, etc, using lable istio: citadel.

**Comment by kyessenov** [comment]
There is no istio-mixer service in helm charts. We have istio-policy and
istio-telemetry that are backed by the same mixer component.

**Comment by myidpt** [comment]
@kyessenov I see your point. But "istio-ca" service itself doesn't work well for the API it is serving. Istio CA API is serving/going to serve:
1. CSR verification and signing,
2. Root certificate distribution,
3. CRL distribution.

And we probably don't want to have each of these features defined as a separate service. These features are not what a CA can do. So we need to change the name "istio-ca" for sure. I think using the name "istio-citadel" works.

**Comment by wattli** [comment]
please remember to update corresponding docs as well.

**Comment by myidpt** [comment]
Yes, I'm working on that.

**Comment by incfly** [comment]
#4805

**Pull request merged**